service/chartconfig/v1/resource/chart/create.go

```diff
@@ -1,7 +1,64 @@
 package chart
 
-import "context"
+import (
+	"context"
+	"fmt"
+
+	"github.com/giantswarm/chart-operator/service/chartconfig/v1/key"
+	"github.com/giantswarm/microerror"
+)
 
 func (r *Resource) ApplyCreateChange(ctx context.Context, obj, createChange interface{}) error {
+	customObject, err := key.ToCustomObject(obj)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+	chartState, err := toChartState(createChange)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	if chartState.ChartName != "" {
+		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("creating chart %s", chartState.ChartName))
+
+		tarballPath, err := r.apprClient.PullChartTarball(customObject)
```

Comment:
Thinking about reusing the appr and helm client here it would make sense to decouple the design from chart-operator specifics like the `ChartConfig` CR. `helmClient.InstallFromTarball` is cool because it takes the path which is not bound to the operator but here the ties are quite heavy.

Reply:
Very good point, done, now the appr methods just accept the name of the chart and the channel.

Reply:
+1 Let's also do this with the helm client next time we change it.